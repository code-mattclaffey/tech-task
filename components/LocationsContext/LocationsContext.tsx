import React, { useState } from "react";

type ProviderProps = {};

const initialContext = {
    searchTerm: "",
    searchResults: [
        {
            city: "Amsterdam",
            country: "NL",
            count: 21301,
            locations: 14
        },
        {
            city: "Badhoevedorp",
            country: "NL",
            count: 2326,
            locations: 1
        },
        {
            city: "Amsterdam",
            country: "NL",
            count: 21301,
            locations: 14
        },
        {
            city: "Badhoevedorp",
            country: "NL",
            count: 2326,
            locations: 1
        },
        {
            city: "Amsterdam",
            country: "NL",
            count: 21301,
            locations: 14
        },
        {
            city: "Badhoevedorp",
            country: "NL",
            count: 2326,
            locations: 1
        }
    ],
    selectedLocations: [
        {
            location: "Punjabi Bagh",
            city: "Delhi",
            country: "IN",
            measurements: [
                {
                    parameter: "so2",
                    value: "7.8",
                    lastUpdated: "2015-07-24T11:30:00.000Z",
                    unit: "µg/m3",
                    sourceName: "Punjabi Bagh",
                    averagingPeriod: {
                        unit: "hours",
                        value: 0.25
                    }
                },
                {
                    parameter: "co",
                    value: 1.3,
                    lastUpdated: "2015-08-18T23:30:00.000Z",
                    unit: "mg/m3",
                    sourceName: "CPCB",
                    averagingPeriod: {
                        unit: "hours",
                        value: 0.25
                    }
                },
                {
                    parameter: "pm25",
                    value: 79,
                    lastUpdated: "2015-10-02T21:45:00.000Z",
                    unit: "µg/m3",
                    sourceName: "CPCB",
                    averagingPeriod: {
                        unit: "hours",
                        value: 0.25
                    }
                }
            ]
        }
    ],
    setSearchTerm: () => null,
    removeSelectedLocation: (location: string) => location
};

export const LocationsContext = React.createContext(initialContext);

export const LocationsProvider: React.FunctionComponent<ProviderProps> = ({
    children
}) => {
    const [state, setState] = useState(initialContext);

    const setSearchTerm: any = ({ currentTarget }: any) => {
        const searchTerm = currentTarget.value;
        setState({ ...state, searchTerm });
    };

    const removeSelectedLocation = (locationName: string) => {
        const { selectedLocations } = state;
        const filteredSelectedLocations = selectedLocations.filter(
            ({ location }) => location !== locationName
        );

        setState({
            ...state,
            selectedLocations: filteredSelectedLocations
        });
    };

    const data = { ...state, setSearchTerm, removeSelectedLocation };

    return (
        // @ts-ignore
        <LocationsContext.Provider value={data}>
            {children}
        </LocationsContext.Provider>
    );
};
