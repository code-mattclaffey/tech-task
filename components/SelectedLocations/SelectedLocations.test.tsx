import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { SelectedLocations } from '.';
import { LocationsProvider } from '../LocationsContext/LocationsContext';

afterEach(() => {
    cleanup();
});

describe('<SelectedLocations />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <LocationsProvider>
                <SelectedLocations />
            </LocationsProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
