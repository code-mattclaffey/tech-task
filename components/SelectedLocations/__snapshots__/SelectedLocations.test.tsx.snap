// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedLocations /> should match snapshot 1`] = `<div />`;

exports[`<SelectedLocations /> should render one selected city 1`] = `
<div>
  <section
    aria-label="Selected locations"
  >
    <ul
      class="c-selected-locations"
    >
      <li
        class="c-selected-locations__location"
      >
        <div
          class="c-card"
          data-testid="selected-location"
        >
          <button
            aria-label="Remove Punjabi Bagh from the selected locations list"
            class="c-button c-button--plain c-button--small c-selected-locations__remove-city-btn"
            data-testid="remove-btn"
            type="button"
          >
            <svg
              fill="none"
              height="30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
          </button>
          <span
            class="c-selected-locations__attr u-text--uppercase u-text--small"
          >
            Updated 4 years ago
          </span>
          <span
            class="c-selected-locations__attr c-selected-locations__attr--heading e-heading e-heading--h6"
          >
            Punjabi Bagh
          </span>
          <span
            class="c-selected-locations__attr"
          >
            in 
            Delhi
            , 
            IN
          </span>
          <span
            class="c-selected-locations__attr u-text--semibold u-text--uppercase"
          >
            <span
              class="u-text--captilize"
            >
              Values:
            </span>
             
            so2: 7.8, co: 1.3, pm25: 79
          </span>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
